buildSrc/subprojects/buildquality/src/main/kotlin/org/gradle/gradlebuild/buildquality/incubation/IncubatingApiReportTask.kt

```diff
@@ -84,17 +117,20 @@ class Analyzer @Inject constructor(
 
     override
     fun run() {
-        val versionToIncubating = mutableMapOf<String, MutableSet<String>>()
+        val versionToIncubating = mutableMapOf<Version, MutableSet<IncubatingDescription>>()
         srcDirs.forEach { srcDir ->
             if (srcDir.exists()) {
-                val solver = JavaSymbolSolver(CombinedTypeSolver(JavaParserTypeSolver(srcDir), ReflectionTypeSolver()))
-                srcDir.walkTopDown().forEach {
-                    if (it.name.endsWith(".java")) {
-                        try {
-                            parseJavaFile(it, versionToIncubating, solver)
-                        } catch (e: Exception) {
-                            println("Unable to parse $it: ${e.message}")
+                val sourceFileParser = CompositeVersionsToIncubatingCollector(listOf(
```

Comment:
❓ Consider renaming the variable?
```suggestion
                val collector = CompositeVersionsToIncubatingCollector(listOf(
```